### PR TITLE
add task queue manager

### PR DESCRIPTION
**Pull request by loning**

### CODE REVIEW

File: AElf.Contracts.TestBase/ContractTester.cs [hunk 1]
```diff
@@ -362,9 +362,17 @@ public async Task<Block> MineAsync(List<Transaction> txs)
             var blockchainService = Application.ServiceProvider.GetRequiredService<IBlockchainService>();
             var preBlock = await blockchainService.GetBestChainLastBlockHeaderAsync();
             var minerService = Application.ServiceProvider.GetRequiredService<IMinerService>();
+            var blockchainExecutingService = Application.ServiceProvider.GetRequiredService<IBlockchainExecutingService>();
 
-            return await minerService.MineAsync(preBlock.GetHash(), preBlock.Height,
+            var block = await minerService.MineAsync(preBlock.GetHash(), preBlock.Height,
                 DateTime.UtcNow.AddMilliseconds(int.MaxValue));
+            
+            await blockchainService.AddBlockAsync(block);
```
Comment (loning) [inline AElf.Contracts.TestBase/ContractTester.cs]: wrap it in a service.

File: AElf.Contracts.TestKit/TransactionExecutor.cs [hunk 2]
```diff
@@ -30,9 +31,15 @@ public async Task ExecuteAsync(Transaction transaction)
             var blockchainService = _serviceProvider.GetRequiredService<IBlockchainService>();
             var preBlock = await blockchainService.GetBestChainLastBlockHeaderAsync();
             var minerService = _serviceProvider.GetRequiredService<IMinerService>();
+            var blockchainExecutingService = _serviceProvider.GetRequiredService<IBlockchainExecutingService>();
 
-            await minerService.MineAsync(preBlock.GetHash(), preBlock.Height,
+            var block = await minerService.MineAsync(preBlock.GetHash(), preBlock.Height,
```
Comment (loning) [inline AElf.Contracts.TestKit/TransactionExecutor.cs]: wrap it in a service.

File: AElf.Core.Tests/TaskQueueTests.cs [hunk 3]
```diff
@@ -76,7 +76,7 @@ public void Test_Dispose()
                 testQueue.Enqueue(async () =>
                 {
                     var value = result;
-                    Task.Delay(500);
+                    await Task.Delay(1000);
```
Comment (loning) [inline AElf.Core.Tests/TaskQueueTests.cs]: no need to wait such long time.

every ci test need to run this code

File: AElf.Kernel.SmartContractExecution/Application/IBlockAttachService.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    public interface IBlockAttachService
+    {
+        void AttachBlock(Block block);
+    }
+
+    public class BlockAttachService : IBlockAttachService, ITransientDependency
+    {      
+        private readonly IBlockchainService _blockchainService;
+        private readonly IBlockchainExecutingService _blockchainExecutingService;
+        private readonly ITaskQueueManager _taskQueueManager;
```
Comment (loning) [inline AElf.Kernel.SmartContractExecution/Application/IBlockAttachService.cs]: I have changed my mind, move queue mananger out of the service.

Comment (zhxymh) [reply AElf.Kernel.SmartContractExecution/Application/IBlockAttachService.cs]: What is your idea and how to use it now?

**Pull request merged**